### ISSUE TITLE
[13.0] GettextAlias._get_cr may choose a closed cr

### ISSUE THREAD

**Issue by ap-wtioit** [odoo/odoo#76559]
Impacted versions:
13.0

Steps to reproduce:
```
if some_errors:
    with registry(self._cr.dbname).cursor() as cr:
        persist_logs(some_errors, cr)
    raise UserError(_(some_errors[0]))
```

Current behavior:
GettextAlias._get_cr picks the first cr found in the frame/stack

Expected behavior:
GettextAlias._get_cr picks the first unclosed cr found in the frame/stack

Video/Screenshot link (optional):

Support ticket number submitted via odoo.com/help (optional):

**Comment by wtaferner** [comment]
@mart-e 
What do you think?

**Comment by mart-e** [comment]
Indeed, we try to retrieve keywords like `cr`, `context`, `self`,.. when calling the `_()` method to be able to retrieve the `ir.translation` records.
If your cursor is closed `GettextAlias` will fail when using it. To fix your issue, you must open a new cursor.

For instance, when translating an error message after an SQL error, we explicitly open a new cursor.
https://github.com/odoo/odoo/blob/f62c20d38844c50c678e4233913a97845420d766/odoo/service/model.py#L76-L84

```py
raise UserError(_(some_errors[0]))
```
This won't work by the way, you should only pass static strings to `_()` if you want [the export](https://github.com/odoo/odoo/blob/14.0/odoo/tools/translate.py#L1150-L1151) to work.
